tests/unittests/libretroshare/gxs/common/data_support.cc

```diff
@@ -209,10 +213,11 @@ RsSerialType* init_item(RsNxsSyncGrpItem& rsgl)
     rsgl.publishTs = rand()%23;
     init_random(rsgl.grpId) ;
 
-    return new RsNxsSerialiser(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
+    if(ser)
+    *ser = new RsNxsSerialiser(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
 }
 
-RsSerialType* init_item(RsNxsSyncMsgItem& rsgml)
+void init_item(RsNxsSyncMsgItem& rsgml,RsSerialType **ser)

```

Comment:
yes. Most of the time we go for small changes (which is good) but at some point these changes may form an inconsistent group of changes, so it needs to be re-written...